### PR TITLE
Add comment to sample

### PR DESCRIPTION
**Pull request by thelander**
### Added

- Add comment to sample (#1440)

### How to prepare for test
- [x] ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] paxa the environment: `paxa`
- [x] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [x] do `cg set sample ACC6449A12 -kv comment "Testkommentar"`

### Expected test outcome
- [x] check that comment has "2022-06-14 10:27-jimmy.thelander: Testkommentar" added as a new line above the original comment if there is one, otherwise check that the same comment is the only comment on the sample
- [x] Take a screenshot and attach or copy/paste the output.

## Review
- [x] code approved by PG
- [x] tests executed by JT
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Deploy this branch
- [ ] Inform to KN, production

### CODE REVIEW

File: tests/cli/set/test_sample.py [hunk 1]
```diff
@@ -194,6 +194,38 @@ def test_priority_number(cli_runner: CliRunner, base_context: CGConfig, base_sto
...
+        sample, [sample_obj.internal_id, "-kv", key, newer_value, "-y"], obj=base_context
+    )
+
+    # THEN it should have newer_value above new_value as attribute key on the sample and in LIMS
+    comments = getattr(sample_obj, key).split("\n")
+    assert result.exit_code == SUCCESS
+    assert comments[0].endswith(newer_value)
+    assert comments[1].endswith(new_value)
+    assert base_context.lims_api.get_updated_sample_key() == key
+    assert base_context.lims_api.get_updated_sample_value() == newer_value
```
Comment (patrikgrenfeldt) [inline tests/cli/set/test_sample.py:226]: I think this test should be split into two tests to get independent test-cases, one that tests setting a comment on a sample without comment and one test that tests setting comment on a sample with an existing comment

Comment (thelander) [reply tests/cli/set/test_sample.py:226]: > I think this test should be split into two tests to get independent test-cases, one that tests setting a comment on a sample without comment and one test that tests setting comment on a sample with an existing comment

I agree. I'll split the test into two separate test cases.

File: cg/cli/set/base.py [hunk 2]
```diff
@@ -266,8 +266,12 @@ def sample(
         if not (yes or click.confirm(CONFIRM)):
             continue
 
-        setattr(sample_obj, new_key, new_value)
-        _update_comment(_generate_comment(new_key, old_value, new_value), sample_obj)
+        if key != "comment":
```
Comment (patrikgrenfeldt) [inline cg/cli/set/base.py:269]: maybe we should swith the order of the if-and else to avoid the negation

File: tests/cli/set/test_sample.py [hunk 3]
```diff
@@ -194,6 +194,49 @@ def test_priority_number(cli_runner: CliRunner, base_context: CGConfig, base_sto
...
+    assert base_context.lims_api.get_updated_sample_key() == key
+    assert base_context.lims_api.get_updated_sample_value() == new_value
+
+
+def test_sample_comment_append(
+    cli_runner: CliRunner, base_context: CGConfig, base_store: Store, helpers
+):
+    # GIVEN a database with a sample with a comment
+    old_value = "test comment"
+    sample_obj = helpers.add_sample(
```
Comment (patrikgrenfeldt) [inline tests/cli/set/test_sample.py:220]: 👍

File: cg/cli/set/base.py (hunk 2 above)
Comment (thelander) [reply cg/cli/set/base.py]: Thanks! I removed the negation in the `if` statement.

**Review by patrikgrenfeldt — APPROVED** [review]
Great!

**Comment by thelander** [comment]
Before:
image

After:
image

**Comment by thelander** [comment]
Deployed to prod

image

**Pull request merged**